tests/functional/object/rgw/test_rgw_virtual_hosted_ops.py

```diff
@@ -0,0 +1,157 @@
+            )
+        else:
+            write_random_objects_in_pod(
+                awscli_pod_session, test_directory_setup.origin_dir, amount=1, bs="50M"
+            )
+            config_cmd = config_cmd + " multipart_chunk_size_mb = 5"
+        return config_cmd
+
+    @pytest.mark.parametrize(
+        argnames="multipart",
+        argvalues=[pytest.param(True), pytest.param(False)],
+        ids=[
+            "multipart-obj",
+            "non-multipart-obj",
+        ],
+    )
+    def test_host_style_obj_upload(
+        self,
+        awscli_pod_session,
+        rgw_bucket_factory,
+        test_directory_setup,
+        multipart,
+    ):
+        """
+        Test s3 operations on virtual style hosted buckets
+
+        """
+        logger.info("Creating obc")
+        bucket_name = rgw_bucket_factory(amount=1, interface="RGW-OC")[0].name
+        config_cmd = self.prepare_cmd_and_data(
+            bucket_name, multipart, awscli_pod_session, test_directory_setup
+        )
+        logger.info("Uploading object to the bucket")
+        awscli_pod_session.exec_sh_cmd_on_pod(
+            command=f"s3cmd {config_cmd} put {test_directory_setup.origin_dir}/ObjKey-0 s3://{bucket_name}"
+        )
+        logger.info("Listing the objects in the bucket")
+        op = awscli_pod_session.exec_sh_cmd_on_pod(
+            command=f"s3cmd {config_cmd} ls s3://{bucket_name}"
+        )
+        logger.info(op)
+
+    @pytest.mark.parametrize(
+        argnames="multipart",
+        argvalues=[pytest.param(True), pytest.param(False)],
+        ids=[
+            "multipart-obj",
+            "non-multipart-obj",
+        ],
+    )
+    def test_host_style_obj_download(
+        self, awscli_pod_session, rgw_bucket_factory, test_directory_setup, multipart
+    ):
+        """
+        Test download operations on virtual style hosted buckets
+
+        """
+        logger.info("Creating obc")
+        bucket_name = rgw_bucket_factory(amount=1, interface="RGW-OC")[0].name
+        config_cmd = self.prepare_cmd_and_data(
+            bucket_name, multipart, awscli_pod_session, test_directory_setup
+        )
+        logger.info("Uploading object to the bucket")
+        awscli_pod_session.exec_sh_cmd_on_pod(
+            command=f"s3cmd {config_cmd} put {test_directory_setup.origin_dir}/ObjKey-0 s3://{bucket_name}"
+        )
+        logger.info("Downloading object from the bucket")
+        awscli_pod_session.exec_sh_cmd_on_pod(
+            command=f"s3cmd {config_cmd} get s3://{bucket_name}/ObjKey-0 {test_directory_setup.result_dir}"
+        )
+        verify_s3_object_integrity(
+            original_object_path=f"{test_directory_setup.origin_dir}/ObjKey-0",
+            result_object_path=f"{test_directory_setup.result_dir}/ObjKey-0",
+            awscli_pod=awscli_pod_session,
+        )
+
+    @pytest.mark.parametrize(
+        argnames="multipart",
+        argvalues=[pytest.param(True), pytest.param(False)],
```

Comment:
and also here

Reply:
Test plan is not yet pushed to Polarion. Will update the test script once I get IDs for these test cases